### DISCUSSION TITLE
Validate/Disable Sensors

### DISCUSSION THREAD

**Discussion by gschnitzler** [Hypfer/Valetudo#709] [Q&A / Support] [answered]
I got a Roborock S6 and rooted it directly after unwrapping, without a 'does it work correctly before voiding the warranty' check.
When the device is in the dock, and i hit the start button, it powers up all its gizmos, moves an inch, and then frantically rotates on the spot for minutes before eventually giving up with 'error 4' (cliff sensors).

When the robot is put away from its dock, it immediately complains about its cliff sensors, without powering anything up.
Tried tile, wood and laminate flooring as well as taping the cliff sensors. Reverted to original firmware and updated it to latest version. I double checked all connectors and also the wheels are working in both directions. No luck.
So... i guess i got a lemon. There are others with similar issues; some of which could fix it by replacing the sensors, others by replacing their pcb. But.. is there maybe a simpler solution? I don't have any cliffs in immediate need for detection....

Could the sensors be disabled in software? Òr the error simply ignored or overridden? 
Any ideas?

**Comment by Hypfer** [comment, marked as answer]
That's beyond the scope of valetudo. Try asking in the dustcloud telegram group

**Reply by gschnitzler** [reply]
Thanks for the reply. I had a look at the inner workings of the robot shortly after posting this and came to same conclusion. 
There is this: https://github.com/arne48/xiaomi_bridge
But that route would mean to completly reimplement the whole firmware or monkey patch it. So, to answer my own question: its infeasible.
